### PR TITLE
Media: Restore validity of media with varied mime types

### LINKED ISSUE #5435: Editor: uploading .m4a files results in 'file type not supported' error
A user in 2670241-t reported that uploading `.m4a` / `.m4v` files to the Editor results in the following error message:

screen shot 2016-05-18 at 2 04 58 pm

> 1 file could not be uploaded because the file type is not supported.

This happens both when the file is uploaded via drag and drop, or via the `Add New` button, and I was able to reproduce this issue on both Mac and Windows, using Chrome / Firefox / Safari.

`.m4a` / `.m4v` files can be uploaded via WP-Admin's Editor or Media Library, and once embedded in a Post, play without a problem. 

The user's site, as well as all test sites I used, had the necessary upgrades to upload and use audio and video.

Also just making a note that this issue is limited to `.m4a` / `.m4v` files. `.mp4` files upload without an issue in the Calypso Editor.

**Steps to reproduce**
1) Make sure your test site has a plan, or a legacy space upgrade (and therefore supports audio/video uploads)
2) Create a new Post / open an existing Post in the Editor.
3) Upload a `.m4a` or `.mf4v` file via drag and drop or the `Add New` button.

Please let me know if you need any other info. Thank you!

### PR DESCRIPTION
**Pull request by aduth**
Fixes #5435

This pull request seeks to resolve the issue described in #5435, where it is not currently possible to upload mp3 or m4v files (and presumably others).

__Implementation notes:__

- Removes handling of blob wrappers from `MediaUtils`, instead imposes that the file object passed is already a reference the original file.
- Changes behavior of URL string handling to check for validity of URL using `valid-url` dependency before parsing as URL. Non-URLs are parsed as file paths.

__Testing instructions:__

Ensure Mocha tests pass:

```
npm run test-client
```

Verify that file uploads continue to succeed for (a) media not historically with any uploading issues, (b) media with special characters in their names, and (c) media with varied mime types.

__Open questions:__

- Should we improve `getFileExtensionFromMimeType` to account for these cases? Currently, it's not very reliable. Additionally, perhaps we should return a value more indicative of a failing case, like `null` or `undefined`, rather than an empty string.
- Is there a case where we'd expect the name to be defined on a blob wrapper at `file.fileName`, but not available from the original raw object `file.fileContents` ?

/cc @gwwar @robobot3000

### CODE REVIEW

**Comment by gwwar** [comment]
@aduth could we add some additional tests for these? It's currently very difficult to detect a regression unless we attempt to upload with all supported file types in testing.

File: client/lib/media/test/utils.js [hunk 1]
```diff
@@ -180,10 +160,6 @@ describe( 'MediaUtils', function() {
 			expect( MediaUtils.getMimeType( 'example#?#?.gif?w=100' ) ).to.be.undefined;
 		} );
 
-		it( 'should ignore querystring parameters', function() {
-			expect( MediaUtils.getMimeType( 'example.gif?w=100' ) ).to.equal( 'image/gif' );
```
Comment (gwwar) [inline client/lib/media/test/utils.js]: Let's update this test to test a full url rather than a fragment.

**Comment by gwwar** [comment]
Reading up on how browsers map mime types, it is safer to parse from the file extension from whatever the file claims to be.

> Should we improve getFileExtensionFromMimeType to account for these cases? Currently, it's not very reliable. Additionally, perhaps we should return a value more indicative of a failing case, like null or undefined, rather than an empty string.

I would probably patch it up for known cases, or remove it as a helper if it's too unreliable to use.

> Is there a case where we'd expect the name to be defined on a blob wrapper at file.fileName, but not available from the original raw object file.fileContents

This is to handle a prototyping case where we create a local Blob but not a File. We cannot set a name property on a Blob, so we wrap it in a plain object wrapper with a fileName property.

**Comment by gwwar** [comment]
Verified that this fixes the Chrome only issue.

**Comment by aduth** [comment]
@gwwar Added two additional tests in 8a7bf7cd58f53203f163c480f3f5dc02fd8d3b22 for verifying results of utility functions when passed string URLs containing query arguments.

**Pull request merged**